### ISSUE TITLE
Release Zeebe 0.11.0

### ISSUE THREAD

**Issue by npepinpe** [zeebe-io/zeebe#924]

**Comment by saig0** [comment]
Client API:
* a user can open a management subscription to receive deployment and topic events/command (#901)

**Comment by Zelldon** [comment]
## Default output mapping behavior
This was changed from replace to top-level merge, regarding to user feedback.
### What does this mean?

#### Before:

Description | Task Payload | Workflow Instance Payload | Output Mapping | Result
-- | -- | -- | -- | --
Replacement| {  "sum": 234.97 } | {  "prices": [    199.99,    29.99,    4.99] } | none | { "sum": 234.97 }

#### Now

Description | Task Payload | Workflow Instance Payload | Output Mapping | Result
-- | -- | -- | -- | --
Merge top level | {  "sum": 234.97 } | {  "prices": [    199.99,    29.99,    4.99] } | none | {  "prices": [    199.99,    29.99,    4.99],  "sum": 234.97 }

### What was the problem with the solution before:

You had to do the top-level merge on your own. If you do not want to replace the complete payload and merge the job result with the workflow instance payload you had to specify each property. See for example this [forum thread](https://forum.zeebe.io/t/change-default-behavior-of-output-mapping-of-tasks/121)

## Output behavior is configurable
Since we changed the default output behavior to merge we thought it can happen that some of the users may wants to overwrite the complete payload. For this and for the use case that the worker may no produce any value we introduce the `outputBehavior` attribute which accepts three different states:

 * **MERGE** merges the job result on top level into the workflow instance payload, if no output mapping is specified. If output mappings are specified then it does a mapping specific merge. This is the default output behavior.
 * **OVERWRITE** overwrites the workflow instance payload with the job result. If output mappings are specified, then the content is extracted from the job result, which then overwrites the workflow instance payload.
 * **NONE** indicates that the worker does not produce any output. It is not possible to use output mappings in combination with this behavior. The Job result will simply dropped.

**Example:**
```xml
<serviceTask id="collectMoney">
    <extensionElements>
      <zeebe:ioMapping outputBehavior="overwrite">
        <zeebe:input source="$.price" target="$.total"/>
        <zeebe:output source="$.paymentMethod" target="$.paymentMethod"/>
       </zeebe:ioMapping>
    </extensionElements>
</serviceTask>
```

Related issues are:

 * https://github.com/zeebe-io/zeebe/issues/907
 * https://github.com/zeebe-io/zeebe/issues/908
 * https://github.com/zeebe-io/zeebe/issues/911
 * https://github.com/zeebe-io/zeebe/issues/930

**Comment by Zelldon** [comment]
## SourceRecordPosition is published by the Broker

Most of the events are produced/written as an follow up event of an event/command, by the stream processors. For these cases the records have the `sourceRecordPosition`, which point the record which caused them. These property was not published to the client before, now it is send on each command response and also the subscribed records contain these field.

This feature enables the clients to understand better, which events causes the follow up events and which are related to each other. It is also possible to visualize these dependencies.

See for example the following diagram. It shows the records of an workflow instance, which had an incident on an XOR-Gateway. The incident was resolved and the workflow instance was completed cleanly.

![shouldresolveincidentforexclusivegatewaywithoutmatchingcondition dot](https://user-images.githubusercontent.com/2758593/41291763-e126bd14-6e50-11e8-98b9-cbead0b836fb.png)

**Comment by npepinpe** [comment]
## Snapshot Replication

Parts were already released in the 0.10.0 release (#876 and #877), and now with #878 the feature is complete.

At the moment, to speed up startup time, brokers can recover their state from a snapshot instead of reprocessing the whole log stream, essentially cutting down how much they have to reprocess.

The problem is when switching leaders, i.e. leader goes away, a new broker becomes leader. This new leader has all the events in its log, no problem, but it didn't have any snapshots, so it has to reprocess everything to build up its internal state, which can take some time. To speed this up, we also replicate snapshots to followers, so that they can quickly become leaders as well, thereby facilitating *fast failover*.

The current strategy is very simple: followers will periodically poll the partition leader for a manifest of the snapshots it has, and will serially download them to its snapshot folder, by recursively fetching each snapshot file in fixed chunks to a temporary file; once finished, it compares the checksum of the manifest to the downloaded file, and if correct, will save the temporary file as a valid snapshot file, writing its companion checksum file as well.
